### PR TITLE
Support for JSON Language Server 

### PR DESCRIPTION
**Pull request by kiennq**
Per #1073, adding `lsp-json` with server automatically installation capability.

### CODE REVIEW

File: lsp-json.el [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+(require 'lsp-mode)
+(require 'ht)
+(require 'url-util)
+
+(defgroup lsp-json nil
+  "LSP support for JSON, using vscode-json-languageserver."
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/vscode-langservers/vscode-json-languageserver")
+  :package-version '(lsp-mode . "6.2"))
```
Comment (yyoncho) [inline lsp-json.el]: 6.2 was just released. We are now 6.3

**Comment by kiennq** [comment]
> 1. Is it possible to associate the current buffer with a specific schema?

Yes, either via `"$schema": "..."` in json file or special mapping with `lsp-json-schemas`.
Please refer [here](https://github.com/vscode-langservers/vscode-json-languageserver#settings) for latter settings.

> 2. Do we need a method like lsp-json-update-schemas which should be called when schema mappings are changed?

Not really, if we update schema then just run `lsp-restart-workspace` would be easier :).
I don't want to provide additional method (we can try to do it through `:set` for `defcustom`) unless it's necessary.

File: lsp-json.el (hunk 1 above)
Comment (kiennq) [reply lsp-json.el]: Updated!

**Comment by yyoncho** [comment]
Ok, if `lsp-json--schema-associations` is intended to be used by the users it should be renamed to `lsp-json-schema-associations`.

**Comment by kiennq** [comment]
> Ok, if `lsp-json--schema-associations` is intended to be used by the users it should be renamed to `lsp-json-schema-associations`.

No, user should use `lsp-json-schemas`, the `lsp-json--schema-associations` is internal.

**Comment by yyoncho** [comment]
Thank you!

**Comment by yyoncho** [comment]
> No, user should use `lsp-json-schemas`, the `lsp-json--schema-associations` is internal.

Yes, you are right.

**Comment by yyoncho** [comment]
Do you have twitter? If yes, can you twit about it with a gif? IMO it will be very useful language server for a lot of people.

**Comment by yyoncho** [comment]
@kiennq one more thing - if you have twitter account it will be great to tweet about the new functionality with a gif. IMO this server will be helpful for a lot of people.

**Comment by kiennq** [comment]
Thanks, here is my twit https://twitter.com/neikmaster/status/1204859535610957824

File: lsp-json.el [hunk 2]
```diff
@@ -0,0 +1,136 @@
+;;; lsp-pwsh.el ---  vscode-json-languageserver integration -*- lexical-binding: t; -*-
```
Comment (thisch) [inline lsp-json.el]: this filename in this comment is wrong

Comment (kiennq) [reply lsp-json.el]: Oh, the downfall of copy/paste :(. Will update it next time I touch this file again.

Comment (thisch) [reply lsp-json.el]: No worries.

**Pull request merged**